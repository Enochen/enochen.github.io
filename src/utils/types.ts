export interface ImagesGraph {
  allFile: AllFile
}
export interface AllFile {
  edges?: EdgesEntity[] | null
}
export interface EdgesEntity {
  node: Node
}
export interface Node {
  relativePath: string
  childImageSharp: ChildImageSharp
}
export interface ChildImageSharp {
  fluid: Fluid
}
export interface Fluid {
  base64: string
  aspectRatio: number
  src: string
  srcSet: string
  sizes: string
}