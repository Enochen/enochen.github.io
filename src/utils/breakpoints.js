export const TABLET_MEDIA_QUERY = "@media only screen and (max-width:768px)"