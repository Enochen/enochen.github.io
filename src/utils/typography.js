import Typography from "typography"
import fairyGateTheme from "typography-theme-fairy-gates"
fairyGateTheme.overrideThemeStyles = ({ rhythm }, options) => ({
  'h1,h2': {
    margin: 0
  }
})
const typography = new Typography(fairyGateTheme)
export const { scale, rhythm, options } = typography
export default typography