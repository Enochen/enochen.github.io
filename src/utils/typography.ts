import Typography, { BaseLine } from "typography"
import { TABLET_MEDIA_QUERY } from "typography-breakpoint-constants"
import verticalRhythm from "compass-vertical-rhythm"
import { CSSPseudos, CSSObject } from "styled-components"

const theme = {
  title: "Dasher",
  baseFontSize: "18px",
  baseLineHeight: 1.56,
  scaleRatio: 2,
  headerFontFamily: ["Source Sans Pro", "Helvetica", "Arial", "sans-serif"],
  bodyFontFamily: ["Source Sans Pro", "Helvetica", "Arial", "sans-serif"],
  headerColor: "hsla(0,0%,0%,0.85)",
  bodyColor: "hsla(0,0%,0%,0.7)",
  headerWeight: 400,
  bodyWeight: 400,
  boldWeight: 700,
  overrideStyles: ({ scale, rhythm }, options) => {
    const vr = verticalRhythm({
      baseFontSize: "16px",
      baseLineHeight: "1.5",
    })

    const styles = {
      "h1, h2": {
        margin: 0,
      },
      a: {
        color: "#DB0E0A",
        textDecoration: "none",
      },
      "a:hover": {
        color: "#6460b8",
      },
      html: {
        backgroundColor: "#f3f3f3",
        justifyContent: "center",
        display: "flex",
      },
      // Mobile styles.
      [TABLET_MEDIA_QUERY]: {
        html: {
          ...vr.establishBaseline(),
        },
      },
    }

    return styles
  },
}

export const scale = (value:number):CSSObject => {
  const temp:BaseLine = typography.scale(value)
  return {fontSize: temp.fontSize, lineHeight: temp.lineHeight}
}

const typography:Typography = new Typography(theme)
export * from "typography-breakpoint-constants"
export const { rhythm, options } = typography
export default typography
