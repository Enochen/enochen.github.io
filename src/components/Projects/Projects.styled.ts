import BackgroundImage from 'gatsby-background-image'
import styled from 'styled-components'
import { rhythm, scale, TABLET_MEDIA_QUERY } from '../../utils/typography'

export const Body = styled.div`
  width: 70%;
  display: flex;
  flex-direction: column;
  padding: ${rhythm(0.5)} ${rhythm(0.8)};
`

const BodySubsection = styled.div`
  margin: ${rhythm(0.3)} 0;
`

export const Name = styled(BodySubsection)`
  ${scale(1)};
  margin-bottom: 0.5rem;
`

export const TagGroup = styled(BodySubsection)`
  flex-wrap: wrap;
  display: flex;
  margin-bottom: 0;
`

export const Desc = styled(BodySubsection)``

export const LinkGroup = styled(BodySubsection)`
  flex-wrap: wrap;
  display: flex;
`
export const Tag = styled.div`
  ${scale(-0.3)};
  padding: 3px 10px;
  margin: 0 0.5rem 0.5rem 0;
  font-weight: 600;
  text-align: center;
  background: #e6e6e6;
  transition: color 1s, background 1s;
`

export const BodyLink = styled.a`
  color: #6460b8;
  text-decoration: none;
  padding: 5px 12px;
  border: 1px solid #6460b8;
  margin-right: 1rem;

  &:hover {
    color: white;
    background: #6460b8;
  }
`

export const ImageLink = styled.a`
  width: 50%;
  display: block;
  overflow: auto;
`

export const Background = styled(BackgroundImage)`
  background-position: top;
  height: 100%;
`

export const Wrapper = styled.div`
  display: flex;
  min-width: 300px;
  margin: 1rem;
  background: white;
  box-shadow: 0 ${rhythm(0.2)} ${rhythm(0.4)} #a2a2a2;
  flex-direction: row-reverse;

  ${TABLET_MEDIA_QUERY} {
    flex-direction: column;
    ${Body} {
      width: 100%;
      box-sizing: border-box;
    }
    ${ImageLink} {
      width: 100%;
      height: 30vh;
    }
  }
`
