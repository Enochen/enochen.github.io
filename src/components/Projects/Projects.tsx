import { graphql, useStaticQuery } from 'gatsby'
import React, { FC } from 'react'
import { ProjectInfo, ProjectList } from '../../data/projectData'
import { ImagesGraph, Fluid } from '../../utils/types'
import { Section } from '../Section'
import * as styled from './Projects.styled'

const ext = {
  target: '_blank',
  rel: 'noopener noreferrer',
}

const findImage = (name: string, imagesGraph: ImagesGraph) => {
  const result = imagesGraph.allFile.edges.find(
    ({ node }) => node.relativePath === name
  )
  if (result) {
    return result.node.childImageSharp.fluid
  }
  return undefined
}

const getFluid = (name: string, imagesGraph: ImagesGraph) => {
  const imageNode = findImage(name, imagesGraph)
  if (imageNode) {
    return imageNode
  }
  return findImage('default.jpg', imagesGraph)
}

interface LinkProps {
  label: string
  url: string
}

const makeTags = (data: string[]) => {
  return data.map(x => <styled.Tag key={x}>{x}</styled.Tag>)
}

const makeLinks = (data: LinkProps[]) => {
  return data.map(x => (
    <styled.BodyLink key={x.label} href={x.url} {...ext}>
      {x.label}
    </styled.BodyLink>
  ))
}

interface ProjectFluid extends ProjectInfo {
  fluid: Fluid
}

export const Project: FC<ProjectFluid> = ({
  name,
  desc,
  tags = [],
  live,
  github,
  fluid,
}) => {
  // Image
  const imageURL = live ? live : github

  // Body
  const tagData = makeTags(tags)
  const links: LinkProps[] = [
    ...(live ? [{ label: 'Project', url: live }] : []),
    ...(github ? [{ label: 'Github', url: github }] : []),
  ]
  const linkData = makeLinks(links)

  return (
    <styled.Wrapper>
      <styled.ImageLink href={imageURL} aria-label={name} {...ext}>
        <styled.Background fluid={fluid}></styled.Background>
      </styled.ImageLink>
      <styled.Body>
        <styled.Name>{name}</styled.Name>
        <styled.TagGroup>{tagData}</styled.TagGroup>
        <styled.Desc dangerouslySetInnerHTML={{ __html: desc }}></styled.Desc>
        <styled.LinkGroup>{linkData}</styled.LinkGroup>
      </styled.Body>
    </styled.Wrapper>
  )
}

export const makeProjects = (data: ProjectInfo[], imagesGraph: ImagesGraph) => {
  return data.map((x, i) => (
    <Project key={i} fluid={getFluid(x.image, imagesGraph)} {...x} />
  ))
}

export const Projects: FC<ProjectList> = ({ data }) => {
  const images: ImagesGraph = useStaticQuery(
    graphql`
      query {
        allFile {
          edges {
            node {
              relativePath
              childImageSharp {
                fluid(quality: 100) {
                  ...GatsbyImageSharpFluid_withWebp
                }
              }
            }
          }
        }
      }
    `
  )
  const projects = makeProjects(data, images)
  return (
    <Section id="projects" title="Projects">
      {projects}
    </Section>
  )
}
