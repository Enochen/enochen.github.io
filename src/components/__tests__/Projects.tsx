import { StaticQuery } from 'gatsby'
import React from 'react'
import renderer from 'react-test-renderer'
import { ImagesGraph } from '../../utils/types'
import { PureProjects } from '../Projects'
describe('Projects Section', () => {
  it('renders correctly', () => {
    const images: ImagesGraph = {
      allFile: {
        edges: [
          {
            node: {
              relativePath: 'askpalette.png',
              childImageSharp: {
                fluid: {
                  base64:
                    'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAATCAYAAACQjC21AAAACXBIWXMAAAsSAAALEgHS3X78AAAChElEQVQ4y5WUS2gTURSG41I3SheKuhELNhU1VqUPTFpqcFEjVVFERBAEcedaEZTUagWLC4WAiI9OFaFYxkWgKgYtauuiNiU+xlDTdKZN0qnYpk0mmVfye++kbaZpC9MD35zLvef881/uzLXl83nMQ2M+y7KMcDiMeDyOiYkJI+dyOZhjuV7bcgU0stksgsEgBgcHwXEcQqGQMVdaZ+6lYaODRCIBURQNB7quG5S6oWFep4wJAiIjI1A1reiQPuj2KLTBDJ0zO6Fj83omk4EkSQsvX7LlUvuBQAAsy8Lv94NhGMRisQWnK/UYDvOkYDkyUhqzySSE0VEIPA9NVUlnaZ2+8qGUBheJoLe/H+FoFB+/9GGabG+lKDqkTsYEZPnoEqZ+/4L4I4R/3E9MkjwbGYYsjC6qyUQjUKanig7VdArDTU4INXbwzt0QDu4y4Anjrj2I1TuMPE7yGFnn59YNXA5EHdsQ890rCOpaQVA45sZk/V6Ihw5AbNxvHXc1EtV2iA/vzwnqsGlE8K+nFum6cqRcdqScFdZpqMRM1VZM+dqLDuW0hBdnL4PxXEBn88VV8fz4JTx2n8PXp12Fz4k6TGU0HL6bxL5WFTW3FVTfsk5tm4Kd12U8eDtjCOo5cihpIui5k0SdV4Xrpgxni2KZhlYFVddk+N4sElRwpm0ATTeGcNQbXBXNLUNovDqAJz2RoqAqz+IPWw6BXQee3UBYbxnhdRmiXWsQ/+YtHEqOHIquELu9ZcAn2+r5TPhgg8pdgfGf5ImgQhy+667E+1dlhE2EjZYJdG9Gz8u1+D7QOndpEMFUVkJN53lsf3YCFR2nsKPjpGUqmdPY8ugI2vsYQ1AjF8V/RR3DxIr061cAAAAASUVORK5CYII=',
                  aspectRatio: 1.079847908745247,
                  src:
                    '/static/e6be3240ed2c270f52af63ccf3c552b2/35dab/askpalette.png',
                  srcSet:
                    '/static/e6be3240ed2c270f52af63ccf3c552b2/a6865/askpalette.png 200w,\n/static/e6be3240ed2c270f52af63ccf3c552b2/e52d5/askpalette.png 400w,\n/static/e6be3240ed2c270f52af63ccf3c552b2/35dab/askpalette.png 800w,\n/static/e6be3240ed2c270f52af63ccf3c552b2/7ef28/askpalette.png 1200w,\n/static/e6be3240ed2c270f52af63ccf3c552b2/088bd/askpalette.png 1420w',
                  sizes: '(max-width: 800px) 100vw, 800px',
                },
              },
            },
          },
          {
            node: {
              relativePath: 'hbhb.png',
              childImageSharp: {
                fluid: {
                  base64:
                    'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAHCAYAAAAIy204AAAACXBIWXMAAAsSAAALEgHS3X78AAAAvElEQVQoz62Q2QqDQAxF0x8TXGr//xtEUEdnwF1Eb7mBFF+6PDRwyE0mE5KIiOBf3OjTNMUn4jhGFEXKu5okSXDPMjzyHIIvtu87iqJAWZb4xeQ4DkzThHmesSyLaouHYcA4jhrzjbkrzK/rqn7bNpznCWFhXdfoug5N06BtW4Wa+aqq4Jx75Qljw3uvNaxnU52Qk1hDen5iIbU14xth3hqHENT3fa8Ts5fekILYWlyD2mLTV+wMvDFXJbQnJ4q5wTMr9dsAAAAASUVORK5CYII=',
                  aspectRatio: 2.6949654491609083,
                  src:
                    '/static/8ca940abf88b8bf0aaae74ad7921f4a9/35dab/hbhb.png',
                  srcSet:
                    '/static/8ca940abf88b8bf0aaae74ad7921f4a9/a6865/hbhb.png 200w,\n/static/8ca940abf88b8bf0aaae74ad7921f4a9/e52d5/hbhb.png 400w,\n/static/8ca940abf88b8bf0aaae74ad7921f4a9/35dab/hbhb.png 800w,\n/static/8ca940abf88b8bf0aaae74ad7921f4a9/7ef28/hbhb.png 1200w,\n/static/8ca940abf88b8bf0aaae74ad7921f4a9/c9c17/hbhb.png 1600w,\n/static/8ca940abf88b8bf0aaae74ad7921f4a9/a1612/hbhb.png 2730w',
                  sizes: '(max-width: 800px) 100vw, 800px',
                },
              },
            },
          },
          {
            node: {
              relativePath: 'cue.jpg',
              childImageSharp: {
                fluid: {
                  base64:
                    'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAbABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAEDBAX/xAAVAQEBAAAAAAAAAAAAAAAAAAAAAf/aAAwDAQACEAMQAAAB7ylE2gE5aVDAr//EAB4QAAIBAwUAAAAAAAAAAAAAAAECAwAQEgQREyAh/9oACAEBAAEFAjXtpY+RV04SzjIYbDp//8QAFBEBAAAAAAAAAAAAAAAAAAAAIP/aAAgBAwEBPwEf/8QAFBEBAAAAAAAAAAAAAAAAAAAAIP/aAAgBAgEBPwEf/8QAHBAAAQQDAQAAAAAAAAAAAAAAAQACEBESICFR/9oACAEBAAY/ApxJpcc72KVA6//EABgQAAMBAQAAAAAAAAAAAAAAAAERIQAQ/9oACAEBAAE/IS0DghEA3NaIsOcCxgLTOHLn/9oADAMBAAIAAwAAABAjybz/xAAVEQEBAAAAAAAAAAAAAAAAAAAQEf/aAAgBAwEBPxAp/8QAFhEBAQEAAAAAAAAAAAAAAAAAEEER/9oACAECAQE/EIZT/8QAHRABAQACAgMBAAAAAAAAAAAAAREAECFBMVGBkf/aAAgBAQABPxBBALcaGj3y6hwinHjBYIdOZgePmiCoPYzBkdU96FMBe/3X/9k=',
                  aspectRatio: 0.75,
                  src: '/static/12f91ede981c5c0f871ac7b5b8969da8/c26dd/cue.jpg',
                  srcSet:
                    '/static/12f91ede981c5c0f871ac7b5b8969da8/471e9/cue.jpg 200w,\n/static/12f91ede981c5c0f871ac7b5b8969da8/dc2cb/cue.jpg 400w,\n/static/12f91ede981c5c0f871ac7b5b8969da8/c26dd/cue.jpg 800w,\n/static/12f91ede981c5c0f871ac7b5b8969da8/150ec/cue.jpg 1200w',
                  sizes: '(max-width: 800px) 100vw, 800px',
                },
              },
            },
          },
          {
            node: {
              relativePath: 'default.jpg',
              childImageSharp: {
                fluid: {
                  base64:
                    'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAANABQDASIAAhEBAxEB/8QAGQAAAgMBAAAAAAAAAAAAAAAAAAQBAgMF/8QAFQEBAQAAAAAAAAAAAAAAAAAAAgH/2gAMAwEAAhADEAAAAX8Yob0hYR//xAAbEAACAwADAAAAAAAAAAAAAAAAAgMREgETIf/aAAgBAQABBQJmoilblrO037s//8QAFBEBAAAAAAAAAAAAAAAAAAAAEP/aAAgBAwEBPwE//8QAFBEBAAAAAAAAAAAAAAAAAAAAEP/aAAgBAgEBPwE//8QAGBAAAwEBAAAAAAAAAAAAAAAAABAxARH/2gAIAQEABj8COaoRf//EABsQAQEBAAIDAAAAAAAAAAAAABEBABAhMWGh/9oACAEBAAE/IQdNtSTDX5wpAnvSTfPRRN//2gAMAwEAAgADAAAAEKw//8QAFhEAAwAAAAAAAAAAAAAAAAAAEBEh/9oACAEDAQE/EFB//8QAFBEBAAAAAAAAAAAAAAAAAAAAEP/aAAgBAgEBPxA//8QAHBABAAMAAwEBAAAAAAAAAAAAAQARITFBUXFh/9oACAEBAAE/EHRTJ2K5UuWpXVjX3ifcLq1M537kAY07v2MNO/s//9k=',
                  aspectRatio: 1.4982332155477032,
                  src:
                    '/static/1ef070b1ed84e0e52d2895ad9de033cf/c26dd/default.jpg',
                  srcSet:
                    '/static/1ef070b1ed84e0e52d2895ad9de033cf/471e9/default.jpg 200w,\n/static/1ef070b1ed84e0e52d2895ad9de033cf/dc2cb/default.jpg 400w,\n/static/1ef070b1ed84e0e52d2895ad9de033cf/c26dd/default.jpg 800w,\n/static/1ef070b1ed84e0e52d2895ad9de033cf/150ec/default.jpg 1200w,\n/static/1ef070b1ed84e0e52d2895ad9de033cf/86ab4/default.jpg 1600w,\n/static/1ef070b1ed84e0e52d2895ad9de033cf/b3045/default.jpg 2120w',
                  sizes: '(max-width: 800px) 100vw, 800px',
                },
              },
            },
          },
        ],
      },
    }
    const projectData = {
      data: [
        {
          name: 'Google',
          desc: 'I made google',
          tags: ['HTML/CSS'],
          live: 'https://google.org',
          github: 'https://github.com/google/google',
          image: 'google.jpg',
        },
        {
          name: 'cue',
          desc: 'right on cue!',
          tags: ['Python'],
          live: 'https://www.cornelldti.org/Projects/events',
          github: 'https://github.com/cornell-dti/events-backend/',
          image: 'cue.jpg',
        },
      ],
    }
    const tree = renderer
      .create(<PureProjects {...projectData} images={images} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
