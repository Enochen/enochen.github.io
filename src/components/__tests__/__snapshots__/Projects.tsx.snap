// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects Section renders correctly 1`] = `
<div
  className="sc-AykKC cyoTwy"
  id="projects"
>
  <div
    className="sc-AykKD dhTwqA"
  >
    <h1
      className="sc-AykKE dZpaAc"
    >
      Projects
    </h1>
  </div>
  <div
    className="sc-fzXfLS dUPWpO"
  >
    <a
      aria-label="Google"
      className="sc-fzXfLQ fweGBN"
      href="https://google.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      <div
        className="sc-fzXfLR kWqkrD gbi-773652653-3oim9gNw2qp9AZy6x1AYt1 gatsby-image-wrapper"
        style={
          Object {
            "background-position": "top",
            "backgroundPosition": "center",
            "backgroundRepeat": "no-repeat",
            "backgroundSize": "cover",
            "opacity": 0.99,
            "position": "relative",
          }
        }
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": "
          .sc-fzXfLR.kWqkrD.gbi-773652653-3oim9gNw2qp9AZy6x1AYt1:before,
          .sc-fzXfLR.kWqkrD.gbi-773652653-3oim9gNw2qp9AZy6x1AYt1:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            
            transition: opacity 0.5s ease 0.25s;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;
backgroundposition: top;

          }
          .sc-fzXfLR.kWqkrD.gbi-773652653-3oim9gNw2qp9AZy6x1AYt1:before {
            z-index: -100;
            
            background-image: url(data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAANABQDASIAAhEBAxEB/8QAGQAAAgMBAAAAAAAAAAAAAAAAAAQBAgMF/8QAFQEBAQAAAAAAAAAAAAAAAAAAAgH/2gAMAwEAAhADEAAAAX8Yob0hYR//xAAbEAACAwADAAAAAAAAAAAAAAAAAgMREgETIf/aAAgBAQABBQJmoilblrO037s//8QAFBEBAAAAAAAAAAAAAAAAAAAAEP/aAAgBAwEBPwE//8QAFBEBAAAAAAAAAAAAAAAAAAAAEP/aAAgBAgEBPwE//8QAGBAAAwEBAAAAAAAAAAAAAAAAABAxARH/2gAIAQEABj8COaoRf//EABsQAQEBAAIDAAAAAAAAAAAAABEBABAhMWGh/9oACAEBAAE/IQdNtSTDX5wpAnvSTfPRRN//2gAMAwEAAgADAAAAEKw//8QAFhEAAwAAAAAAAAAAAAAAAAAAEBEh/9oACAEDAQE/EFB//8QAFBEBAAAAAAAAAAAAAAAAAAAAEP/aAAgBAgEBPxA//8QAHBABAAMAAwEBAAAAAAAAAAAAAQARITFBUXFh/9oACAEBAAE/EHRTJ2K5UuWpXVjX3ifcLq1M537kAY07v2MNO/s//9k=);
            opacity: 0; 
          }
          .sc-fzXfLR.kWqkrD.gbi-773652653-3oim9gNw2qp9AZy6x1AYt1:after {
            z-index: -101;
            background-image: url(data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAANABQDASIAAhEBAxEB/8QAGQAAAgMBAAAAAAAAAAAAAAAAAAQBAgMF/8QAFQEBAQAAAAAAAAAAAAAAAAAAAgH/2gAMAwEAAhADEAAAAX8Yob0hYR//xAAbEAACAwADAAAAAAAAAAAAAAAAAgMREgETIf/aAAgBAQABBQJmoilblrO037s//8QAFBEBAAAAAAAAAAAAAAAAAAAAEP/aAAgBAwEBPwE//8QAFBEBAAAAAAAAAAAAAAAAAAAAEP/aAAgBAgEBPwE//8QAGBAAAwEBAAAAAAAAAAAAAAAAABAxARH/2gAIAQEABj8COaoRf//EABsQAQEBAAIDAAAAAAAAAAAAABEBABAhMWGh/9oACAEBAAE/IQdNtSTDX5wpAnvSTfPRRN//2gAMAwEAAgADAAAAEKw//8QAFhEAAwAAAAAAAAAAAAAAAAAAEBEh/9oACAEDAQE/EFB//8QAFBEBAAAAAAAAAAAAAAAAAAAAEP/aAAgBAgEBPxA//8QAHBABAAMAAwEBAAAAAAAAAAAAAQARITFBUXFh/9oACAEBAAE/EHRTJ2K5UuWpXVjX3ifcLq1M537kAY07v2MNO/s//9k=);
            
            
          }
        ",
            }
          }
        />
        <noscript>
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": "
    .sc-fzXfLR.kWqkrD.gbi-773652653-3oim9gNw2qp9AZy6x1AYt1:before {
      opacity: 1;
      background-image: url('/static/1ef070b1ed84e0e52d2895ad9de033cf/c26dd/default.jpg');
    }",
              }
            }
          />
        </noscript>
      </div>
    </a>
    <div
      className="sc-AykKF hukWXa"
    >
      <div
        className="sc-AykKG sc-AykKH Opbog"
      >
        Google
      </div>
      <div
        className="sc-AykKG sc-AykKI klVsbl"
      >
        <div
          className="sc-fzXfLO bpdGYk"
        >
          HTML/CSS
        </div>
      </div>
      <div
        className="sc-AykKG sc-AykKJ gtgQKK"
        dangerouslySetInnerHTML={
          Object {
            "__html": "I made google",
          }
        }
      />
      <div
        className="sc-AykKG sc-AykKK soPFw"
      >
        <a
          className="sc-fzXfLP dvxyfQ"
          href="https://google.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          Project
        </a>
        <a
          className="sc-fzXfLP dvxyfQ"
          href="https://github.com/google/google"
          rel="noopener noreferrer"
          target="_blank"
        >
          Github
        </a>
      </div>
    </div>
  </div>
  <div
    className="sc-fzXfLS dUPWpO"
  >
    <a
      aria-label="cue"
      className="sc-fzXfLQ fweGBN"
      href="https://www.cornelldti.org/Projects/events"
      rel="noopener noreferrer"
      target="_blank"
    >
      <div
        className="sc-fzXfLR kWqkrD gbi--1810794872-4awbyijmSKcBDc9BVyxqWm gatsby-image-wrapper"
        style={
          Object {
            "background-position": "top",
            "backgroundPosition": "center",
            "backgroundRepeat": "no-repeat",
            "backgroundSize": "cover",
            "opacity": 0.99,
            "position": "relative",
          }
        }
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": "
          .sc-fzXfLR.kWqkrD.gbi--1810794872-4awbyijmSKcBDc9BVyxqWm:before,
          .sc-fzXfLR.kWqkrD.gbi--1810794872-4awbyijmSKcBDc9BVyxqWm:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            
            transition: opacity 0.5s ease 0.25s;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;
backgroundposition: top;

          }
          .sc-fzXfLR.kWqkrD.gbi--1810794872-4awbyijmSKcBDc9BVyxqWm:before {
            z-index: -100;
            
            background-image: url(data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAbABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAEDBAX/xAAVAQEBAAAAAAAAAAAAAAAAAAAAAf/aAAwDAQACEAMQAAAB7ylE2gE5aVDAr//EAB4QAAIBAwUAAAAAAAAAAAAAAAECAwAQEgQREyAh/9oACAEBAAEFAjXtpY+RV04SzjIYbDp//8QAFBEBAAAAAAAAAAAAAAAAAAAAIP/aAAgBAwEBPwEf/8QAFBEBAAAAAAAAAAAAAAAAAAAAIP/aAAgBAgEBPwEf/8QAHBAAAQQDAQAAAAAAAAAAAAAAAQACEBESICFR/9oACAEBAAY/ApxJpcc72KVA6//EABgQAAMBAQAAAAAAAAAAAAAAAAERIQAQ/9oACAEBAAE/IS0DghEA3NaIsOcCxgLTOHLn/9oADAMBAAIAAwAAABAjybz/xAAVEQEBAAAAAAAAAAAAAAAAAAAQEf/aAAgBAwEBPxAp/8QAFhEBAQEAAAAAAAAAAAAAAAAAEEER/9oACAECAQE/EIZT/8QAHRABAQACAgMBAAAAAAAAAAAAAREAECFBMVGBkf/aAAgBAQABPxBBALcaGj3y6hwinHjBYIdOZgePmiCoPYzBkdU96FMBe/3X/9k=);
            opacity: 0; 
          }
          .sc-fzXfLR.kWqkrD.gbi--1810794872-4awbyijmSKcBDc9BVyxqWm:after {
            z-index: -101;
            background-image: url(data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAbABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAEDBAX/xAAVAQEBAAAAAAAAAAAAAAAAAAAAAf/aAAwDAQACEAMQAAAB7ylE2gE5aVDAr//EAB4QAAIBAwUAAAAAAAAAAAAAAAECAwAQEgQREyAh/9oACAEBAAEFAjXtpY+RV04SzjIYbDp//8QAFBEBAAAAAAAAAAAAAAAAAAAAIP/aAAgBAwEBPwEf/8QAFBEBAAAAAAAAAAAAAAAAAAAAIP/aAAgBAgEBPwEf/8QAHBAAAQQDAQAAAAAAAAAAAAAAAQACEBESICFR/9oACAEBAAY/ApxJpcc72KVA6//EABgQAAMBAQAAAAAAAAAAAAAAAAERIQAQ/9oACAEBAAE/IS0DghEA3NaIsOcCxgLTOHLn/9oADAMBAAIAAwAAABAjybz/xAAVEQEBAAAAAAAAAAAAAAAAAAAQEf/aAAgBAwEBPxAp/8QAFhEBAQEAAAAAAAAAAAAAAAAAEEER/9oACAECAQE/EIZT/8QAHRABAQACAgMBAAAAAAAAAAAAAREAECFBMVGBkf/aAAgBAQABPxBBALcaGj3y6hwinHjBYIdOZgePmiCoPYzBkdU96FMBe/3X/9k=);
            
            
          }
        ",
            }
          }
        />
        <noscript>
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": "
    .sc-fzXfLR.kWqkrD.gbi--1810794872-4awbyijmSKcBDc9BVyxqWm:before {
      opacity: 1;
      background-image: url('/static/12f91ede981c5c0f871ac7b5b8969da8/c26dd/cue.jpg');
    }",
              }
            }
          />
        </noscript>
      </div>
    </a>
    <div
      className="sc-AykKF hukWXa"
    >
      <div
        className="sc-AykKG sc-AykKH Opbog"
      >
        cue
      </div>
      <div
        className="sc-AykKG sc-AykKI klVsbl"
      >
        <div
          className="sc-fzXfLO bpdGYk"
        >
          Python
        </div>
      </div>
      <div
        className="sc-AykKG sc-AykKJ gtgQKK"
        dangerouslySetInnerHTML={
          Object {
            "__html": "right on cue!",
          }
        }
      />
      <div
        className="sc-AykKG sc-AykKK soPFw"
      >
        <a
          className="sc-fzXfLP dvxyfQ"
          href="https://www.cornelldti.org/Projects/events"
          rel="noopener noreferrer"
          target="_blank"
        >
          Project
        </a>
        <a
          className="sc-fzXfLP dvxyfQ"
          href="https://github.com/cornell-dti/events-backend/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Github
        </a>
      </div>
    </div>
  </div>
</div>
`;
